drake/examples/Pendulum/runEnergyShaping.cpp

```diff
@@ -7,14 +5,11 @@
 using namespace std;
 
 int main(int argc, char* argv[]) {
-  Pendulum p;
   shared_ptr<lcm::LCM> lcm(new lcm::LCM);
   if(!lcm->good())
     return 1;
 
-  p.input_frame = make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumInput",p.input_frame->getCoordinateNames(),lcm);
-  p.output_frame = make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumState",p.output_frame->getCoordinateNames(),lcm);
-
+  Pendulum p(lcm);
   DrakeSystemPtr controller = make_shared<PendulumEnergyShaping>(p);
```

Comment:
I'm not worried about the overhead of creating the variable. I just think we need to not use shared_ptr blindly everywhere and stay aware of ownership, which I believe my nagging comments have accomplished :-)